pkg/slo-controller/noderesource/plugins/batchresource/plugin.go

```diff
@@ -225,6 +233,11 @@ func getNodeMetricUsage(info *slov1alpha1.NodeMetricInfo) corev1.ResourceList {
 func getNodeAllocatable(node *corev1.Node) corev1.ResourceList {
 	result := node.Status.Allocatable.DeepCopy()
 	result = quotav1.Mask(result, []corev1.ResourceName{corev1.ResourceCPU, corev1.ResourceMemory})
+
+	// if something reserved by node.annotation, should remove those resource.
```

Comment:
The node reserved resources would be substracted in the `Calculate()`. We should not double-subtract the node reserved.

Reply:
this function is used in `Calculate()`

Reply:
Yes, it is used in the `Calculate`. Please check the formula if the `nodeAllocatable` was subtracted with the `node.anno.reserved`.

Reply:
o, got it

Reply:
`nodeAllocatable` contains  node.allocatable[memory] and node.allocatable[cpu], we haven't modified this, so there is no double-subtract `node.anno.reserved`

Reply:
We've defined the formula: `batch.alloc := node.alloc * ratio - sum(pod(Prod).usage) - max(sys.usage, node.anno.reserved)`, where node.anno.reserved would be subtracted no more than once. But in the current implementation, we subtract the node.anno.reserved firstly in the `getNodeAllocatable` and secondly in the `Calculate`.

Reply:
node.alloc = node.alloc - node.anno.reserved
batch.alloc := node.alloc * ratio - sum(pod(Prod).usage) - sys.usage
so use this formula to calculate?

Reply:
> node.alloc = node.alloc - node.anno.reserved batch.alloc := node.alloc * ratio - sum(pod(Prod).usage) - sys.usage so use this formula to calculate?

This formula is incorrect as we discussed in [#922](https://github.com/koordinator-sh/koordinator/pull/922#discussion_r1063255945), where sys.usage also includes the usage of node.anno.reserved.

Reply:
done